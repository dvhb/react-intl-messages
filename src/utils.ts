import * as fs from 'fs';
import * as G from 'glob';
import * as https from 'https';
import * as querystring from 'querystring';

export const glob = (pattern: string, ignore?: string | string[]): Promise<string[]> =>
  new Promise((resolve, reject) => {
    G(pattern, { ignore }, (err, val) => (err ? reject(err) : resolve(val)));
  });

export const readFile = (file: string): Promise<string> =>
  new Promise((resolve, reject) => {
    fs.readFile(file, 'utf8', (err, data) => (err ? reject(err) : resolve(data)));
  });

export const writeFile = (file: string, contents: string): Promise<void> =>
  new Promise((resolve, reject) => {
    fs.writeFile(file, contents, 'utf8', err => (err ? reject(err) : resolve()));
  });

export const posixPath = (fileName: string) => fileName.replace(/\\/g, '/');

type Options = https.RequestOptions & {
  url: string;
  body?: object;
  qs?: object;
};
export const request = <T>({ url, body, qs, headers = {}, ...rest }: Options) => {
  const newUrl = qs ? `${url}?${querystring.stringify(qs)}` : url;

  const bodyString = JSON.stringify(body);

  if (body) {
    headers['Content-Length'] = Buffer.byteLength(bodyString, 'utf8');
  }

  return new Promise<T>((resolve, reject) => {
    const request = https.request(newUrl, { headers, ...rest }, response => {
      let data = '';
      response.on('data', chunk => {
        data += chunk;
      });
      response.on('end', () => resolve(JSON.parse(data)));
    });

    if (body) {
      request.write(bodyString);
    }

    request.on('error', err => {
      reject(err);
    });
    request.end();
  });
};

const format = (time: Date) => time.toTimeString().replace(/.*(\d{2}:\d{2}:\d{2}).*/, '$1');
export const showError = (message: string) => console.error('\x1b[31m', message, '\x1b[0m');
export const showInfo = (message: string) => console.info(`\x1b[34m[${format(new Date())}]\x1b[0m`, `${message}`);
