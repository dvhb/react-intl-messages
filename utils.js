const { promisify } = require('util');
const { readFile, writeFile } = require('fs');
const globPkg = require('glob');
const request = require('request');

const glob = (pattern, ignore) =>
  new Promise((resolve, reject) => {
    globPkg(pattern, { ignore }, (err, val) => (err ? reject(err) : resolve(val)));
  });

module.exports = {
  request: promisify(request),
  readFile: promisify(readFile),
  writeFile: promisify(writeFile),
  glob,
};
