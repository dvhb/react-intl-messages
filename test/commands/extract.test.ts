import { test } from '@oclif/test';
import { readJson, mkdir, rmdir } from '../utils';

beforeAll(() => {
  mkdir('messages');
});

afterAll(() => {
  rmdir('messages');
});

describe('extract', () => {
  test
    .stdout()
    .command(['extract'])
    .it('extract messages into en.json', () => {
      expect(readJson('messages/en.json')).toMatchSnapshot();
    });
});
