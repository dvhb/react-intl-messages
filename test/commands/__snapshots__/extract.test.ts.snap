// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extract extract messages into en.json 1`] = `
Array [
  Object {
    "defaultMessage": "Hello {name}, you have {unreadCount, number} {unreadCount, plural, one {message} other {messages}}",
    "files": Array [
      "src/App.js",
    ],
    "id": "welcome",
    "message": "",
  },
]
`;

exports[`extract runs extract 1`] = `
Array [
  Object {
    "defaultMessage": "Hello {name}, you have {unreadCount, number} {unreadCount, plural, one {message} other {messages}}",
    "files": Array [
      "src/App.js",
    ],
    "id": "welcome",
    "message": "",
  },
]
`;
